### PR TITLE
[Admin][UI][API] Request password reset validation

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12          |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | -                      |
| License         | MIT                                                          |

Based on the work done in #14138 

Added validation for the admin's request password reset feature.

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Action/Account/RequestPasswordResetAction.php [hunk 1]
```diff
@@ -42,24 +45,30 @@ public function __invoke(Request $request): RedirectResponse
...
-            ));
+            return new RedirectResponse($this->router->generate($redirectRoute));
         }
 
-        return new RedirectResponse($this->router->generate($redirectRoute));
+        return new Response(
+            $this->twig->render('@SyliusAdmin/Security/requestPasswordReset.html.twig', [
+                'form' => $form->createView(),
+            ])
+        );
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/AdminBundle/Action/Account/RequestPasswordResetAction.php:72]: Why did you change the previous behavior 🤔?

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Bundle/AdminBundle/Action/Account/RequestPasswordResetAction.php (hunk 1 above)
Comment (NoResponseMate) [reply src/Sylius/Bundle/AdminBundle/Action/Account/RequestPasswordResetAction.php:72]: To render the validation errors. Pretty sure you can't do that with a redirect. I could just put them in an error flash, but that seems messy.

File: src/Sylius/Behat/Context/Api/Admin/ResettingPasswordContext.php [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        $this->request->updateContent(['newPassword' => $password]);
+    }
+
+    /**
+     * @When I confirm my new password as :password
+     */
+    public function iConfirmMyNewPassword(string $password): void
+    {
+        $this->request->updateContent(['confirmNewPassword' => $password]);
+    }
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ResettingPasswordContext.php]: `When` steps should be defined before `Then` ones

**Review by GSadee — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jan! :tada:
